drivers/serial/uart_ns16550.c

```diff
@@ -657,6 +664,16 @@ static void uart_ns16550_irq_tx_enable(const struct device *dev)
 {
 	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
 
+#ifdef CONFIG_PM
+	struct uart_ns16550_dev_data *const dev_data = dev->data;
+
+	for (int i = ARRAY_SIZE(pm_states) - 1; i >= 0; i--) {
```

Comment:
Suggest leaving a comment why we iterate backwards.

Reply:
This is just to do constraint set, so the loop can be iterated forward.  Fixed.